src/Abstractions/Entities/TaskEntity.cs

```diff
@@ -15,3 +18,117 @@ public interface ITaskEntity
+        {
+            if (parameter.HasDefaultValue)
+            {
+                input = parameter.DefaultValue;
+                return true;
+            }
+
+            input = null;
+            return false;
+        }
+
+        input = operation.GetInput(parameter.ParameterType);
+        return true;
+    }
+
+    bool TryDispatchMethod(TaskEntityOperation operation, out object? result)
+    {
+        static void CheckDuplicateBinding(
+            ParameterInfo? existing, ParameterInfo parameter, string bindingConcept, TaskEntityOperation operation)
+        {
+            if (existing is not null)
+            {
+                throw new InvalidOperationException($"Error dispatching {operation} to '{parameter.Member}'.\n" +
+                    $"Unable to bind {bindingConcept} to '{parameter}' because it has " +
+                    $"already been bound to parameter '{existing}'. Please remove the duplicate parameter in method " +
+                    $"'{parameter.Member}'.\nEntity operation: {operation}.");
+            }
+        }
+
+        Type t = this.GetType();
+        MethodInfo? method = t.GetMethod(operation.Name, InstanceBindingFlags);
+        if (method is null)
+        {
+            result = null;
+            return false;
+        }
+
+        if (method.ReturnType.IsTaskOrValueTask())
+        {
+            throw new InvalidOperationException($"Error dispatching {operation} to '{method}'.\n" +
+                $"{typeof(Task)} and {typeof(ValueTask)} return values are not supported. Return type found: " +
+                $"{method.ReturnType}.");
+        }
+
+        ParameterInfo[] parameters = method.GetParameters();
+        object?[] inputs = new object[parameters.Length];
+
+        int i = 0;
+        ParameterInfo? inputResolved = null;
+        ParameterInfo? contextResolved = null;
+        ParameterInfo? operationResolved = null;
+        foreach (ParameterInfo parameter in parameters)
+        {
+            if (parameter.ParameterType == typeof(TaskEntityContext))
+            {
+                CheckDuplicateBinding(contextResolved, parameter, "context", operation);
+                inputs[i] = operation.Context;
+                contextResolved = parameter;
+            }
+            else if (parameter.ParameterType == typeof(TaskEntityOperation))
+            {
+                CheckDuplicateBinding(operationResolved, parameter, "operation", operation);
+                inputs[i] = operation;
+                operationResolved = parameter;
+            }
+            else if (TryGetInput(parameter, operation, out object? input))
+            {
+                CheckDuplicateBinding(inputResolved, parameter, "input", operation);
+                inputs[i] = input;
+                inputResolved = parameter;
+            }
+            else
+            {
+                throw new InvalidOperationException($"Error dispatching {operation} to '{method}'.\n" +
```

Comment:
Can we remove the "Either the type is not supported or" part? I don't understand what you are referring to there.

Reply:
Re-did this error message and the logic for it.